app/coffee/WebsocketController.coffee

```diff
@@ -202,6 +203,23 @@ module.exports = WebsocketController =
 		Utils.getClientAttributes client, ["user_id", "project_id"], (error, {user_id, project_id}) ->
 			return callback(error) if error?
 			return callback(new Error("no project_id found on client")) if !project_id?
+
+			updateSize = JSON.stringify(update).length
```

Comment:
Although JSON.stringify is very fast this is a very hot code path, when we have done even a bit more processing here before we do see a decent uptick in cpu usage, what if we pushed the check into `DocumentUpdaterManager.queueChange` where we already stringify the update and do some sanity checks on the data looking for null bytes.

Reply:
I did that in a previous iteration, but thought we want to reject a super large payload asap.
E.g. `update.op` could be very large array which produces a very busy loop here:
https://github.com/overleaf/real-time/blob/c3d76760a6f347b81bdcc8b1426265e074017d8d/app/coffee/WebsocketController.coffee#L238-L241

Reply:
> do some sanity checks on the data looking for null bytes.

I fell for that too: This is a memory corruption check, not an actual check for null bytes in the update.
-> https://github.com/overleaf/document-updater/pull/110#issuecomment-580297820

Reply:
I would tend to trust Henry's judgement on performance issues here. He's usually been right in my experience... If it's not harmful to reject the update later I would do that.

I think it's better for the loop to be slower just in the case of huge documents, than to stringify on every request, even small ones.

Reply:
Alright, I will move the check.

Reply:
Done in 8127b75